packages/components/src/query-controls/types.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import type { Key } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { FormTokenFieldProps } from '../form-token-field/types';
+import type { TreeSelectProps } from '../tree-select/types';
+
+export type Entity = {
+	id: number;
+	name: string;
+	parent?: number | null;
+	value?: string;
+};
+
+export type TermsWithChildren = Array< Entity & { children: Entity[] } >;
+
+export type CategorySelectProps = {
+	categoriesList: Entity[];
+	key: Key;
+	label: TreeSelectProps[ 'label' ];
+	noOptionLabel: TreeSelectProps[ 'noOptionLabel' ];
+	onChange: TreeSelectProps[ 'onChange' ];
+	selectedCategoryId?: Entity[ 'id' ];
+};
+
+export type AuthorSelectProps = {
+	authorList?: Entity[];
```

Comment:
`authorsList` is actually much bigger than the `Entity` type, but we don't use most of its properties, like `_links` and `avatar_urls`:

<img width="965" alt="Screen Shot 2022-12-21 at 4 29 52 PM" src="https://user-images.githubusercontent.com/4063887/209026015-e6943587-7678-4beb-bb5e-d5fd307ce8e8.png">